abort-pending-builds-based-on-string-searched-on-logs.groovy

```diff
@@ -0,0 +1,38 @@
+findItems(jenkins.model.Jenkins.instance.items.findAll());
+
+def findItems(items) {
+  for (item in items) {
+    switch(item) {
+        case org.jenkinsci.plugins.workflow.multibranch.WorkflowMultiBranchProject:
+            item.getItems()?.each { itemMultibranch ->
+                searchStringInLogBuild(itemMultibranch, "String that you are looking for")
+            }
+            break;
+        case org.jenkinsci.plugins.workflow.job.WorkflowJob:
+            searchStringInLogBuild(item, "String that you are looking for")
+            break;
+
+        // you can add other kinds of Items using a case clause, or even a default value.
+
+        case com.cloudbees.hudson.plugins.folder.Folder:
+            findItems(item.getItems());
+            break;
+    }
+  }
+}
```

Comment:
`jenkins.model.Jenkins.instance.items.findAll()` will find every item. So when we later do `item.getItems()` for Multibranch and Folders we are duplicating the work for the child items..

You can simply use the following to get all `Job`s only:

```
jenkins.model.Jenkins.get().allItems(hudson.model.Job).each { item ->
    searchStringInLogBuild(item, "String that you are looking for")
}
```